bazel/BUILD.boost

```diff
@@ -0,0 +1,275 @@
+cc_library(
+    name = "all",
+    hdrs = glob(
+        ["boost/**/*"],
+        exclude = ["boost/**/*.cpp"],
+    ),
+    include_prefix = "boost",
+    includes = ["boost"],
+    strip_include_prefix = "boost",
+    visibility = ["//visibility:public"],
+    deps = [],
+)
+
+cc_library(
+    name = "context",
+    srcs = [
+        "libboost_context.a",
+        "libboost_context.so.1.70.0",
+    ],
+    hdrs = glob(
+        [
+            "boost/context.hpp",
+            "boost/context/**/*.h",
+            "boost/context/**/*.hpp",
+        ],
+    ),
+    visibility = ["//visibility:public"],
+    deps = [
+        ":all",
+    ],
+)
+
+genrule(
```

Comment:
It looks like there's a fair amount of duplication in the various `gen_rules` for boost. Can we create a wrapper like https://github.com/istio-ecosystem/authservice/blob/master/bazel/bazel.bzl?